fs/minfuse/runfs.go

```diff
@@ -0,0 +1,142 @@
+package minfuse
+
+import (
+	"errors"
+	"flag"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+
+	_ "net/http/pprof"
+
+	"github.com/scootdev/fuse"
+	"github.com/scootdev/scoot/fs/min"
+	"github.com/scootdev/scoot/snapshots"
+)
+
+type Options struct {
+	src          string
+	mountpoint   string
+	strategyList []string
+	async        bool
+	trace        bool
+	threadUnsafe bool
+	maxReadahead uint32
+}
+
+func SetupLog() {
+	log.SetFlags(log.LstdFlags | log.Lmicroseconds)
+}
+
+func InitFlags() (*Options, error) {
+	// Note on serial vs threadpool option:
+	// Serial means that no locking will be done at all and we will serve from a single goroutine.
+	// Threadpool means that we will lock where necessary and serve from multiple goroutines.
+	src := flag.String("src_root", "", "source directory to mirror")
+	mountpoint := flag.String("mountpoint", "", "directory to mount at")
+	trace := flag.Bool("trace", false, "whether to trace execution")
+	serveStrategy := flag.String("serve_strategy", "",
+		"Options are any of async|sync;serial|threadpool;readahead_mb=N. Default is 'async;serial;readahead_mb=4'")
+
+	flag.Parse()
+	if *src == "" || *mountpoint == "" {
+		return nil, errors.New("Both src_root and mountpoint must be set")
+	}
+
+	opts := Options{
+		src:          *src,
+		mountpoint:   *mountpoint,
+		trace:        *trace,
+		strategyList: strings.Split(*serveStrategy, ";"),
+		async:        true,
+		threadUnsafe: true,
+		maxReadahead: uint32(4 * 1024 * 1024),
+	}
+	for _, strategy := range opts.strategyList {
+		switch {
+		case strategy == "async":
+			opts.async = true
+		case strategy == "sync":
+			opts.async = false
+		case strategy == "serial":
+			opts.threadUnsafe = true
+		case strategy == "threadpool":
+			opts.threadUnsafe = false
+		case strings.HasPrefix(strategy, "readahead_mb="):
+			readaheadStr := strings.Split(strategy, "=")[1]
+			if readahead, err := strconv.ParseFloat(readaheadStr, 32); err == nil {
+				opts.maxReadahead = uint32(readahead * 1024 * 1024)
+				break
+			}
+			fallthrough
+		default:
+			log.Fatal("Unrecognized strategy", strategy)
+		}
+	}
+	log.Print(opts)
+	return &opts, nil
+}
+
+func Runfs(opts *Options) {
+	snap := snapshots.NewFileBackedSnapshot(opts.src, "only")
```

Comment:
I'd move the snapshot creation to main.go, and then have this function take a Snapshot.

(To use the terminology, a Snapshot is a dependency of minfs, and so we should inject the dependency instead of having the fs create it)